plugins/versionlock.py

```diff
@@ -155,13 +157,25 @@ def run(self):
                 cmd = self.opts.subcommand
 
         if cmd == 'add':
-            _write_locklist(self.base, self.opts.package, self.opts.raw, True,
-                            "\n# Added locks on %s\n" % time.ctime(),
-                            ADDING_SPEC, '')
+            (entry, entry_cmd) = _search_locklist(self.opts.package)
+            if entry == '':
+                _write_locklist(self.base, self.opts.package, self.opts.raw, True,
+                                "\n# Added lock on %s\n" % time.ctime(),
+                                ADDING_SPEC, '')
+            elif cmd != entry_cmd:
+                raise dnf.exceptions.Error(CONFLICTING_SPEC % ('excluded: ' + entry))
```

Comment:
This will not work with translations - the 'excluded' (and the 'locked' string below) will remain untranslated. I would prefer to have two complete strings (to make it clearer for translators) - 'Package {} is already locked' and 'Package {} is already excluded'. Or may be even more explicit - 'Cannot lock already excluded package {}'. Please rephrase it as you need (I'm not native speaker).

Reply:
Thanks, I've redone this as you suggested.

Reply:
One other minor detail: I didn't apply lstrip('!') to entry in the error message when it's an exclusion record. Personally, I prefer to see a raw version of an entry string. But you may find that less user-friendly.